### PR TITLE
Bump golangci-lint. Use golangci-lint to perform gosec and gofmt checks

### PR DESCRIPTION
**Pull request by guillaumerose**
Minor changes around the linter:
* Use golangci-lint to run gosec and gofmt
* Bump golangci-lint and fix some issues. 3 gosec issues are not trivial to fix, I disabled them for the moment.
* Remove dead code around the global state (found by gosec - easier to delete it)

Log of gosec errors (goos linux):

```
pkg/extract/extract.go:111:17: G110: Potential DoS vulnerability via decompression bomb (gosec)
			if _, err := io.Copy(file, tarReader); err != nil {
			             ^
pkg/extract/extract.go:171:16: G110: Potential DoS vulnerability via decompression bomb (gosec)
		if _, err := io.Copy(targetFile, fileReader); err != nil {
		             ^
pkg/crc/state/global.go:46:11: G306: Expect WriteFile permissions to be 0600 or less (gosec)
	if err = ioutil.WriteFile(state.FilePath, jsonData, 0644); err != nil {
	         ^
pkg/crc/preflight/preflight_checks_linux.go:324:9: G204: Subprocess launched with function call as argument or cmd arguments (gosec)
	cmd := exec.Command("virsh", "--connect", "qemu:///system", "net-destroy", libvirt.DefaultNetwork)
	       ^
pkg/crc/preflight/preflight_checks_linux.go:326:8: G204: Subprocess launched with function call as argument or cmd arguments (gosec)
	cmd = exec.Command("virsh", "--connect", "qemu:///system", "net-undefine", libvirt.DefaultNetwork)
	      ^

```

### CODE REVIEW

**Comment by gbraad** [comment]
add to whitelist

**Comment by guillaumerose** [comment]
thanks @gbraad !

File: cmd/crc-embedder/cmd/embed.go [hunk 1]
```diff
@@ -86,19 +86,19 @@ func embedFiles(binary string, filenames []string) error {
 
 var (
 	dataFileUrls = map[string][]string{
-		"darwin": []string{
+		"darwin": {
```
Comment (cfergeau) [inline cmd/crc-embedder/cmd/embed.go:89]: This bit is not mentioned in the commit log?

File: .golangci.yml [hunk 2]
```diff
@@ -6,3 +6,4 @@ linters:
   enable:
     - goimports
     - gofmt
+    - gosec
```
Comment (cfergeau) [inline .golangci.yml:9]: There was some discussion around this in https://github.com/code-ready/crc/pull/568 and https://github.com/code-ready/crc/issues/563 I don't know if this is still relevant

File: cmd/crc-embedder/cmd/embed.go (hunk 1 above)
Comment (guillaumerose) [reply cmd/crc-embedder/cmd/embed.go:89]: `make fmtcheck` was not complaining on this but golangci-lint yes. I will add a note in the commit message.

File: .golangci.yml (hunk 2 above)
Comment (guillaumerose) [reply .golangci.yml:9]: Oh I didn't know. 

The main argument was `gosec linter which is part of golangci-lint coming from github.com/golangci/gosec instead github.com/securego/gosec https://github.com/golangci/golangci-lint/blob/master/pkg/golinters/gosec.go#L11 , which is last updated on 2nd Feb https://github.com/golangci/gosec/tree/02.2019 ( https://github.com/golangci/golangci-lint/blob/master/go.mod#L23) which means this is out dated.` but today it's fixed. golangci-lint use the upstream gosec and the latest version.

Comment (praveenkumar) [reply .golangci.yml:9]: So new version of golang-ci does have fix for gosec and now it's taking upstream codebase instead own downstream bits.

File: .travis.yml [hunk 3]
```diff
@@ -6,7 +6,6 @@ go:
 script:
 - make
 - make vendorcheck
-- make fmtcheck
-- make gosec
-- make cross
 - make cross-lint
+- make cross
+- make test
```
Comment (praveenkumar) [inline .travis.yml:11]: :+1:

**Pull request merged**